### DISCUSSION TITLE
Advice for using mise on projects where collaborators use asdf?

### DISCUSSION THREAD

**Discussion by KyleKing** [jdx/mise#3736] [Q&A] [answered]
With the recent mise update, there are stricter errors when a tool can't be found, such as pipx-installed Python dependencies

As an example, this is the .tool-versions file

```
textract-py3 2.0.1
```

Which generates a lot of noisy mise errors with:

```
mise ERROR [~/Developer/project/.tool-versions] textract-py3@2.0.1: textract-py3 not found in mise tool registry
mise ERROR error resolving versions
mise ERROR Run with --verbose or MISE_VERBOSE=1 for more information
```

I have a `mise.toml` in the same directory that contains:

```toml
[tools]
"pipx:textract-py3" = "2.0.1"
```

But mise still errors. Shouldn't `mise.toml` take precedence?

Replacing `.tool-versions` with `mise.toml` isn't viable because of the high switching cost to change our tooling right now. I was the one that migrated us to asfd in the first place when I joined before mise existed

**Edit**: just updated mise and looks like the errors are gone even if mise.toml doesn't exist (and the new feature for converting `.tool-versions` to mise's config 🎉)

**Comment by jdx** [comment, marked as answer]
> But mise still errors. Shouldn't mise.toml take precedence?

no because `pipx:textract-py3` is not the same as `textract-py3`. It's possible different backends could provide different tools, e.g.: `npm:foo` and `pipx:foo` so they can't override.

I don't have a great suggestion other than just ignoring [.tool-versions](https://mise.jdx.dev/configuration/settings.html#override_tool_versions_filename) or the [tool](https://mise.jdx.dev/configuration/settings.html#disable_tools).

**Reply by KyleKing** [reply]
Thanks! TIL about `disable_tools`, which addresses my needs

```sh
mise settings disable_tools textrat-py3,extract-msg
```
